code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/integration/local/LocalAuthPasswordValidator.java

```diff
@@ -0,0 +1,71 @@
+package io.cattle.platform.iaas.api.auth.integration.local;
+
+import io.cattle.platform.archaius.util.ArchaiusUtil;
+import io.cattle.platform.json.JsonMapper;
+import io.github.ibuildthecloud.gdapi.exception.ClientVisibleException;
+import io.github.ibuildthecloud.gdapi.util.ResponseCodes;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.http.HttpResponse;
+import org.apache.http.client.fluent.Request;
+import org.apache.http.entity.ContentType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.netflix.config.DynamicStringProperty;
+
+public class LocalAuthPasswordValidator {
+
+    public static final DynamicStringProperty AUTH_VALIDATE_URL = ArchaiusUtil.getString("api.auth.local.validate.url");
+
+    final static Logger log = LoggerFactory.getLogger(LocalAuthPasswordValidator.class);
+
+    public static void validatePassword(String password, JsonMapper jsonMapper) {
+        String authValidateUrl = AUTH_VALIDATE_URL.get();
+        if (StringUtils.isBlank(authValidateUrl)) {
+            return;
+        }
+
+        Map<String, String> data = new HashMap<String, String>();
+        data.put("secret", password);
+        String jsonString = "";
+        Integer code;
+        HttpResponse response = null;
+
+        try {
+            jsonString = jsonMapper.writeValueAsString(data);
+        } catch (IOException e) {
+            log.error("Error in creating json for POST request", e);
+        }
+
+        try {
+            response =  Request.Post(authValidateUrl).bodyString(jsonString, ContentType.APPLICATION_JSON).execute().returnResponse();
+        } catch (IOException e) {
+            log.error("Error sending POST request", e);
+        }
+
+        code = response.getStatusLine().getStatusCode();
+        if (code >=400 && code <= 499) {
+            Map<String, Object> jsonData = new HashMap<String, Object>();
+            try {
+                jsonData = jsonMapper.readValue(response.getEntity().getContent());
+            } catch (IOException e) {
+                log.error("No JSON response from validator", e);
+            }
+
+            if (!jsonData.containsKey("type") || !jsonData.containsKey("message")) {
```

Comment:
So I added the check for `type` to make sure that the json format returned is correct. And @vincent99 had said in lack of the json message key I should return a 5xx error